### DISCUSSION TITLE
Live AMA - Come fly with Copilot

### DISCUSSION THREAD

**Discussion by tuves** [community/community#29932] [Copilot Conversations]
![GH Copilot](https://user-images.githubusercontent.com/49915749/185383814-28aee88f-daa3-4d68-adf1-0b35faccbe9b.png)
As mentioned in this [post](https://github.com/orgs/community/discussions/29189), on August 22, 2022, the trial period for GitHub CoPilot will end.

On August 24, 2022, we will be hosting an AMA, featuring @iamcjwilliams + @blackgirlbytes to answer all your Copilot questions. We'll have the AMA open for 24 hours to ensure global participation. No need to register--just join in on _this_ discussion, and ask away!

If you have specific [feedback/feature requests](https://github.com/community/community), please share them in the appropriate product categories
If you have questions re: our product roadmap, check out our [public roadmap](https://github.com/github/roadmap)
Please keep questions/comments clean and friendly, refer to our [Code of Conduct](https://github.com/community/community/blob/e76c2ecc5883c65c889caef6d9e63c99c09eeb53/CODE_OF_CONDUCT.md) + [Community Guidelines](https://docs.github.com/articles/github-community-guidelines) if you have any concerns

**Comment by pmnlla** [comment]
surprised no one commented on this yet

**Reply by ettaboyle** [reply]
Thanks for starting the party 🎉

**Reply by pmnlla** [reply]
no problem! looking forward to this 👀

**Reply by Kingofprussia135** [reply]
Thank you for the survey, your a good person.Honestlyandtrulythankyou👀

**Comment by GavinRay97** [comment]
Is this the place to post questions for the AMA, or will there be a link to another discussion tomorrow?

If it is, I have two burning questions:
- I assume Copilot uses the contents of the current project (IE, all files) as contextual information to offer suggestions. Is this the case?
- Is there any way to "train" Copilot on extra data, or ask it to reference some folder(s) for extra context.
   - For instance, it could be useful to point it to the source code of a very large library when working with it another project, so it has more data about how to use the API.

**Reply by ettaboyle** [reply]
This is the spot! Thanks for sharing your questions. The team will be responding starting tomorrow. 🕵🏻

**Reply by tuves** [reply]
Thank you for the questions @GavinRay97! We'll make sure to answer these :)

**Reply by Akash1134** [reply]
Hi @GavinRay97 ,

Great questions! Our Copilot expert @davecheney helped us with answers to your questions.

> I assume Copilot uses the contents of the current project (IE, all files) as contextual information to offer suggestions. Is this the case?

Yes, copilot looks at what we call "related tabs", for example .c files are often paired with .h files, so copilot considers the contents of other tabs open in the editor to try to feed a more complete prompt to the model.

> Is there any way to "train" Copilot on extra data, or ask it to reference some folder(s) for extra context.
> For instance, it could be useful to point it to the source code of a very large library when working with it another project, so it has more data about how to use the API.

No, at the moment this behaviour is not user configurable.

**Reply by GavinRay97** [reply]
Thanks!

**Comment by pmnlla** [comment]
Might be a bit of a dumb question but I have done absolutely zero research on copilot so here goes: 

- To what extent is it currently capable of producing code? 
    - Can it only do single function snippets, write simple scripts, or can it already build full fledged apps? 
- How have you used copilot internally? (preferably in fields where sharing wouldn't violate an NDA)

**Reply by blackgirlbytes** [reply]
Hey there! 
Great questions.
Copilot draws context from comments and code, and suggests individual lines and whole functions instantly. It currently doesn't build full fledged apps. I've personally used Copilot for demos. My role as a dev advocate sometimes includes doing live demos, and it helps me to remember the next lines of code pretty quickly. I've also used it to write documentation as its great at synthesizing while writing documentation. And I've also used it for writing unit tests and regex. It's great at picking up patterns.

Perhaps @iamcjwilliams will have more to add!

**Reply by brandoncjung** [reply]
do you happen to have a good example of the documentation that it helped you write?

**Reply by blackgirlbytes** [reply]
I don't have examples from myself 'cause those were just personal projects in private repositories that I'm not ready to share, but I do have examples from other folks on Twitter. The three tweets below are examples from folks who used Copilot to write documentation and took a screenshot of it. 
- https://twitter.com/MikeMcQuaid/status/1542434802430873600
- https://twitter.com/unixterminal/status/1544790854216437762
- https://twitter.com/manekinekko/status/1537836386614640643

**Reply by pmnlla** [reply]
If copilot can help with documentation **and** code, then i'd honestly consider it a seriously and frequently viable part of my toolkit, as i can't for the life of me be bothered to write documentation if it isn't code comments.

**Comment by 1501575808** [comment]
github的copilot如何退费，八月二十号一直在找退费的入口没找到，昨天八月二十三号下午突然扣费信用卡五百多元，

**Reply by tuves** [reply]
Hi @1501575808,

The best place to go for answers to account-specific questions like this one is by opening a ticket on our [Support page](https://support.github.com/contact). Please open a ticket with them :)

**Comment by tuves** [comment]
The Copilot AMA is opening in 1 minute, folks! Please drop your question(s) into a comment and we'll do our best to answer as many as we can!

**Comment by queenofcorgis** [comment]
How often does Copilot get trained on public repositories?

**Reply by iamcjwilliams** [reply]
The current version was trained on public repositories last, we are planning to ship a new version soon with more recent repos.

**Reply by queenofcorgis** [reply]
Thanks @iamcjwilliams :octocat:

**Comment by ettaboyle** [comment]
**Question for the CoPilot team:** Do we plan to allow customization of colors in Copilot suggestions?

**Reply by iamcjwilliams** [reply]
This would be awesome! The copilot team will jot this down as a feature request

**Comment by ghostinhershell** [comment]
Hey guys 👋
What is the best way to self troubleshoot issues with Copllot?

**Reply by blackgirlbytes** [reply]
If Copilot is not generating any code or comments, I would try installing and reinstalling the extension and refreshing your IDE. Once you've determined that it is working, but if you feel it's not generating the suggestions you were looking for, here's my advice:
- Copilot works best with comments. Write clear, specific, understandable comments for the best results.
- You can also more than just the initial suggestion that Copilot has given you by hovering over the initial suggestion and choosing Previous, Next, or Open Copilot!

**Comment by queenofcorgis** [comment]
A common question from folks: Do we plan to offer an adjusted price for those in countries with low purchasing power?

**Reply by iamcjwilliams** [reply]
Currently, the pricing structure is set: If you choose a monthly billing cycle, you will be billed $10 per calendar month. If you choose a yearly billing cycle, you will be billed $100 per year.

**Comment by LumitoLuma** [comment]
**Question**: Do you have plans to add support for Qt creator? I use it for everything, and it would be great to have a Copilot plug-in.

**Reply by blackgirlbytes** [reply]
Thank you for bringing this to our attention! I'm sure @iamcjwilliams or @smalllemur may know more details around if support is coming.

**Comment by Akash1134** [comment]
Hi there, 👋 Popular question: Is there a way to disable only the current suggestion from Copilot so that I can see the suggestions provided by the IDE?

**Reply by SmallLemur** [reply]
Hi @Akash1134  Thanks for the question ! Currently Copilot can be disabled completely in the copilot menu. We are considering more flexible ways of "muting" Copilot.

**Reply by favourwright** [reply]
> Hi there, 👋 Popular question: Is there a way to disable only the current suggestion from Copilot so that I can see the suggestions provided by the IDE?

What you should do when the "undesired" suggestion is hinted by copilot is press the ``ESC`` key

**Comment by Akash1134** [comment]
Thank you everyone for participating in our Copilot AMA! Though this AMA is now closed, please feel free to direct your questions/feedback to our [Copilot category](https://www.google.com/url?q=https://github.com/orgs/community/discussions/categories/copilot&sa=D&source=calendar&ust=1661872023272270&usg=AOvVaw2RqjfbhY_naojchkV6B5hH). Until the next Community AMA!  🚀
